src/welcomePage.ts

```diff
@@ -14,14 +19,16 @@ export class WelcomePage {
     }
 
     public checkAndShow() {
-        if (!this.context.globalState.get(Constants.IsAzureIoTToolsWelcomePageShown)) {
+        if (this.needToShow()) {
             this.show();
-            this.context.globalState.update(Constants.IsAzureIoTToolsWelcomePageShown, true);
+            TelemetryClient.sendEvent(Constants.AzureIoTToolsShowWelcomePagetEvent, { trigger: "auto" });
+            this.context.globalState.update(Constants.IsAzureIoTToolsWelcomePageShown, extensionVersion);
```

Comment:
IsAzureIoTToolsWelcomePageShown: the naming is not matched with its actual usage.  For anyone without context, they would think it as a boolean value

Reply:
Use AzureIoTToolsWelcomePageLatestVersion instead